src/ClientData/include/ClientData/CaptureData.h

```diff
@@ -288,7 +289,9 @@ class CaptureData {
  private:
   orbit_grpc_protos::CaptureStarted capture_started_;
 
-  orbit_client_data::ProcessData process_;
+  // The initialization is delayed, but still in the constructor. Afterwards, it
+  // must never be std::nullopt.
+  std::optional<orbit_client_data::ProcessData> process_;
```

Comment:
You could use a free function or a lambda to do the initialization during the member initialization phase. That would avoid this being an optional and takes away some complexity.

Reply:
(Feel free to ignore)

Reply:
Done.